### PR TITLE
feat(CX-3233): Use BSV, NSV template for all rejection emails

### PR DESCRIPTION
**Pull request by lordkiz**
This PR resolves [CX-3233]

### Description
- With this PR, all rejection emails will now default to the BSV, NSV template.

[CX-3233]: https://artsyproduct.atlassian.net/browse/CX-3233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/services/submission_service.rb [hunk 1]
```diff
@@ -291,20 +291,8 @@ def deliver_rejection_notification(submission_id)
...
-        when 'Artist Submission'
-          'artist_submission_rejected'
-        when 'NSV', 'BSV'
-          'nsv_bsv_submission_rejected'
-        when 'Not Target Supply'
-          'non_target_supply_artist_rejected'
-        else
-          'other_submission_rejected'
-        end
-
```
Comment (MrSltun) [inline app/services/submission_service.rb:307]: **Question:** Was this change intentional? (couldn't find context on the ticket)

File: app/views/user_mailer/nsv_bsv_submission_rejected.html.erb [hunk 2]
```diff
@@ -11,7 +11,11 @@
         </tr>
         <tr>
           <td class='email-sub-title'>
-            <p>Dear <%= @submission.name %>,</p>
+            <% if @submission.user_name %>
+              <p>Hi <%= @submission.user_name %>,</p>
```
Comment (MrSltun) [inline app/views/user_mailer/nsv_bsv_submission_rejected.html.erb:15]: **Issue:** I believe the final draft has "Dear <name>," instead of "Hi <name>,"

File: app/views/user_mailer/nsv_bsv_submission_rejected.html.erb [hunk 3]
```diff
@@ -11,7 +11,11 @@
...
+            <% else %>
+              <p>Hi there,</p>
+            <% end %>
             <p>Thank you for submitting <%= @submission.title %> by <%= @artist.name %> to us for evaluation. </p>
             <p>Unfortunately, we don’t have a selling opportunity for this work right now. However, we have uploaded it to <%= link_to 'My Collection', 'https://www.artsy.net/settings/my-collection', target: :_blank %> in Artsy to help you track demand.</p>
```
Comment (MrSltun) [inline app/views/user_mailer/nsv_bsv_submission_rejected.html.erb:20]: **Suggestion:** Should we use https://www.artsy.net/my-collection as the URL to open the app if the user has the app?

**Review by MrSltun — COMMENTED** [review]
Great work Kizito! 👏 👏  I just left a few comments :)

File: app/services/submission_service.rb (hunk 1 above)
Comment (MrSltun) [reply app/services/submission_service.rb:307]: nvm, just saw the title of the PR 😂

File: app/views/user_mailer/nsv_bsv_submission_rejected.html.erb (hunk 3 above)
Comment (lordkiz) [reply app/views/user_mailer/nsv_bsv_submission_rejected.html.erb:20]: Yesssss. Thank youuu!

File: app/views/user_mailer/nsv_bsv_submission_rejected.html.erb (hunk 2 above)
Comment (lordkiz) [reply app/views/user_mailer/nsv_bsv_submission_rejected.html.erb:15]: The collector services team opted to use `Hi` instead. See here: https://artsy.slack.com/archives/C01B2P6LJUU/p1671465006706159?thread_ts=1671463083.624579&cid=C01B2P6LJUU

**Review by MrSltun — APPROVED** [review]
🌟 🌟

**Review by olerichter00 — APPROVED** [review]

File: app/mailers/user_mailer.rb [hunk 4]
```diff
@@ -120,7 +120,10 @@ def nsv_bsv_submission_rejected(submission:, artist:)
             unique_args: {
               submission_id: submission.id
             }
-    mail(to: submission.email, subject: 'An update about your submission')
+    title = submission.title ||= 'Unknown'
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply app/mailers/user_mailer.rb:123]: Thank you very much @olerichter00 . You are correct `||` is enough. Infact my original intention was to assign 'Unknown' to the title so that the title matches the subject when we use submission.title in the email body. But with your suggestion I now realised I had already assigned @submission before even setting title. 🤦 . I will fix this in a new PR now.

[the remark this answers is not in the document]
Comment (olerichter00) [reply app/mailers/user_mailer.rb:123]: I guess it's fine and `||=` would work as well. But could lead to hard-to-find bugs.

[the remark this answers is not in the document]
Comment (lordkiz) [reply app/mailers/user_mailer.rb:123]: Yesss. True, It could cause bugs that will be extremely difficult to find. 🙌 . Will change now.

[the remark this answers is not in the document]
Comment (olerichter00) [reply app/mailers/user_mailer.rb:123]: thanks

[the remark this answers is not in the document]
Comment (lordkiz) [reply app/mailers/user_mailer.rb:123]: @olerichter00 Addressed here: https://github.com/artsy/convection/pull/1375 . Thank you!
